apollo-client/src/test/java/com/ctrip/framework/apollo/internals/AbstractConfigTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2021 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.internals;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.ctrip.framework.apollo.ConfigChangeListener;
+import com.ctrip.framework.apollo.enums.ConfigSourceType;
+import com.ctrip.framework.apollo.enums.PropertyChangeType;
+import com.ctrip.framework.apollo.model.ConfigChange;
+import com.ctrip.framework.apollo.model.ConfigChangeEvent;
+import com.google.common.util.concurrent.SettableFuture;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.Test;
+import org.mockito.Matchers;
+
+/**
+ * @author wxq
+ */
+public class AbstractConfigTest {
+
+  /**
+   * @see AbstractConfig#fireConfigChange(ConfigChangeEvent)
+   */
+  @Test
+  public void testFireConfigChange_cannot_notify() {
+    AbstractConfig abstractConfig = spy(new ErrorConfig());
+    final String namespace = "app-namespace-0";
+    ConfigChangeListener configChangeListener = spy(new ConfigChangeListener() {
+      @Override
+      public void onChange(ConfigChangeEvent changeEvent) {
+
+      }
+    });
+    abstractConfig.addChangeListener(configChangeListener, Collections.singleton("cannot-be-match-key"));
+
+    Map<String, ConfigChange> changes = new HashMap<>();
+    changes.put("key1", new ConfigChange(namespace, "key1", null, "new-value", PropertyChangeType.ADDED));
+    ConfigChangeEvent configChangeEvent = new ConfigChangeEvent(namespace, changes);
+
+    assertEquals(0, abstractConfig.fireConfigChange(configChangeEvent));
```

Comment:
I think we can add a counter in this onChange method to test the times of invocation instead of changing the signature